new/supermarket_namespace_governance.md

```diff
@@ -0,0 +1,167 @@
+
+Some languages, by convention, use a special character for this
+purpose. 'alice/json' vs 'bob/json' is a popular method. Enabling this
+requires coordination among the ecosystem tooling at every level  and
+implementations vary wildly. It would be worth exploring the logistics
+of around this, if for no other reason than to have something to point
+to when people suggest it.
+
+Without a special character, semantic strategies fall on a spectrum.
+At one end there is 'json' vs 'json2', or 'json-alice' vs 'json-bob'.
+At the other end, you have... Ruby. While it may be perfectly
+acceptable in the Ruby community to name an XML processing library
+after a Japanese tree saw, I suspect that the Chef community might
+prefer something less random. A governance activity might include the
+creation and enforcement of a recommended naming conventions.
+   
+Some languages have the notion of "core modules", where different
+governance rules come into play. This feels like something that Chef
+should adopt.
+
+This model starts to show cracks when you consider that most Chef
+cookbooks currently on the supermarket are not written in a reusable
+manner. Cookbooks that provide resources can be considered libraries
+or modules, but recipes are really opinionated state policies.
+
+### Chef as a distribution
+
+In this model, we draw parallels with Linux and Unix distributions.
+Most distributions can be described as the set of available packages
+and the metadata that binds them.
+
+Namespace governance is typically optimized for content curation, with
+an emphasis on interoperability. To achieve this, packaging policies
+are created to enforce common conventions.
+
+This model is often used by sophisticated Chef shops, who often write
+everything themselves and rarely consume cookbooks from the public
+Supermarket.
+   
+In Linux distributions, two packages are rarely, if ever, concerned
+with the same software. Rather than having 'alice/apache' vs
+'bob/apache', the distribution will pick a preferred package and
+declare if "the way of the system". Often these packages vary wildly
+between distributions. The configuration conventions around the Apache
+httpd server is a great example of this.
+
+   
+Distributions almost always have a notion of "core packages", much the
+same as many languages. Alternative or additional packages are
+typically used by enabling extra repository sources. Some  systems
+have quazi-official repositories where packages outside the "core" set
+are maintained, and usually still follow the packaging policy for the
+distribution. The RHEL core/EPEL/SCL systems are examples of this, as
+are the Ubuntu universe / PPA systems.
+
+Governing the Supermarket like a Linux distribution might ultimately
+provide the better experience for consumers, but it requires, by an
+order of magnitude, more effort and thought. Interestingly, the
+current flat namespace and dependency tooling fit this model well.
+      
```

Comment:
I think avoiding that is @someara's goal in engaging the community with this RFC, and that the full text adequately expounds upon that risk.

As a consumer of cookbooks, when I need to install, say, Java and Maven, I want straightforward cookbooks within my reach which are fairly cross-platform, reasonably tested, and maintained by someone making an effort not to break my usage in the future.

As a contributor to cookbooks whose goals sometimes are or are not shared by what one might call 'mainline' cookbook maintainers, perhaps toward ensuring the above, I may want an easy way to published a forked or completely alternative cookbook without pointing at a github repository.  I'm just saying that I think the Ubuntu PPA model works well for that, and I think might address the sort of namespacing that e.g. @coderanger wants to see with 'poise' cookbooks'

There doesn't necessarily have to be 'one true way', but there can be a default way and facilitation of alternatives.